src/py/mat3ra/made/tools/build/interface/builders.py

```diff
@@ -270,36 +272,74 @@ class CommensurateLatticeTwistedInterfaceBuilder(BaseBuilder):
     def _generate(self, configuration: _ConfigurationType) -> List[_GeneratedItemType]:
         film = configuration.film
         # substrate = configuration.substrate
-        max_search = self.build_parameters.max_repetition_int
         a = film.lattice.vector_arrays[0][:2]
         b = film.lattice.vector_arrays[1][:2]
-        commensurate_lattice_pairs = self.__generate_commensurate_lattices(
-            configuration, a, b, max_search, configuration.twist_angle
+        max_int = self.build_parameters.max_repetition_int or self.__get_initial_guess_for_max_int(
+            film, configuration.twist_angle
         )
+        commensurate_lattice_pairs: List[CommensurateLatticePair] = []
+        while not commensurate_lattice_pairs and max_int < 42:
+            commensurate_lattice_pairs = self.__generate_commensurate_lattices(
+                configuration, a, b, max_int, configuration.twist_angle
+            )
+            max_int += 1
+
         return commensurate_lattice_pairs
 
+    def __get_initial_guess_for_max_int(self, film, target_angle: float) -> int:
+        """
+        Determine the maximum integer for the transformation matrices based on the target angle.
+
+        Args:
+            a (List[float]): The a lattice vector.
+            b (List[float]): The b lattice vector.
+            target_angle (float): The target twist angle, in degrees.
+
+        Returns:
+            int: The maximum integer for the transformation matrices.
+        """
+        a = film.lattice.vector_arrays[0][:2]
+        b = film.lattice.vector_arrays[1][:2]
+        length_a = np.linalg.norm(a)
+        length_b = np.linalg.norm(b)
+        average_length = (length_a + length_b) / 2
+        theta_rad = np.radians(target_angle)
+
+        # Factor in both the lattice length and twist angle, ensuring a minimum of 1 for very small angles
+        if theta_rad == 0:
+            max_int = 1
+        if film.lattice.type == "HEX":
+            from .enums import angle_to_lmpq_hex
```

Comment:
<img src="https://media2.giphy.com/media/NeDuhuEngZ5ny/giphy.gif"/>

Reply:
?
There's a tabulation for supercell vectors, they themselves do not help construct both matrices in our method, but we can use those values to generate the matrices with max_int value.